New Detection for Multi Cloud

4R9UN opened the pull request:
The detection is centered around Azure and AWS cloud environments, with a primary focus on identifying activities related to IAAS resource abuse within both AWS and Azure cloud environments.

Required items, please complete

Change(s):

New Detection
Reason for Change(s):

Multi-cloud Detection creation
Version Updated:
-yes

Testing Completed:
-yes

4R9UN committed:
New Detection for Multi Cloud 

The detection is centered around Azure and AWS cloud environments, with a primary focus on identifying activities related to IAAS resource abuse within both AWS and Azure cloud environments.

4R9UN committed:
Update ExtensiveAWSComputeResurseDeployements.yaml

4R9UN reviewed (commented):
fix KQL error

4R9UN commented:
@petebryan , Added Cross Cloud detection, kindly review.

4R9UN committed:
Update ExtensiveAWSComputeResurseDeployements.yaml

Query Modified similer like https://microsoft.visualstudio.com/WDATP/_git/InE.MtpCorrelationRules?path=/Detector/Kusto/Rules/AlertDetection/StreamingTrigger/MassVMScaleSetCreationGPU/query.kql&_a=contents&version=GBmaster 
logic

4R9UN reviewed (commented):
edit logic based on below reference 

https://microsoft.visualstudio.com/WDATP/_git/InE.MtpCorrelationRules?path=/Detector/Kusto/Rules/AlertDetection/StreamingTrigger/MassVMScaleSetCreationGPU/query.kql&_a=contents&version=GBmaster

4R9UN committed:
Update SucessfullPosswordSprayAttampOnAWSConsolelogin.yaml

Edit Typo Error

4R9UN committed:
User impersonation Detection Added

4R9UN committed:
Update UserImpersonateByAAID.yaml

4R9UN committed:
Update UserImpersonateByRiskyUser.yaml

4R9UN reviewed (commented):
Error Fix

petebryan commented on Detections/MultipleDataSources/BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml:
```diff
@@ -0,0 +1,106 @@
+  // Exclude entries with empty OriginalRequestId
+      | where isnotempty(OriginalRequestId)
+  // Limit the time range to the last 1 day
+      | where TimeGenerated >= ago(1d)
```
Given this is the same as the query period, you really don't need this in query body itself.

4R9UN replied:
removed the time period

petebryan commented on Detections/MultipleDataSources/BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml:
```diff
@@ -0,0 +1,106 @@
+id: 1f40ed57-f54b-462f-906a-ac3a89cc90d4
+name: Cross-Cloud Credential Brute Force Attempt
```
What about also doing a Cross-Cloud password spray detection?
You could use similar logic to this but with a spray, or also look at instances where a spray is attempted across both AWS and Azure but is only sucessful in one of them.

4R9UN replied:
@petebryan,Title been changed as you recommended, 
logic changed for the spray attempted but expecting more noise as Azure has more number of usage then AWS.

petebryan commented on Detections/MultipleDataSources/BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml:
```diff
@@ -0,0 +1,106 @@
+  // Joining the Azure_Bruforce and AWS_Bruforce tables on matching IP addresses and UserAgents
+  Azure_Bruforce
+    |join kind=inner AWS_Bruforce on $left.IPAddress == $right.SourceIpAddress and UserAgent
+entityMappings:
```
I would consider using alertDetailsOverride and customDetails here to further customize these alerts with details such as impacted account and user name, and also to project out key non-entities such as User Agent.

ref: https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide

4R9UN replied:
done

petebryan commented on Detections/MultipleDataSources/BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml:
```diff
@@ -0,0 +1,106 @@
+          AuthenticationRequirement,
+          ClientAppUsed
+  // Filtering based on conditions for failed and successful events
+      | where FailedEventsCount >= 5
```
Seems odd to have these a different between Azure and AWS - is there a reason for this?

4R9UN replied:
This approach follows the Auch process, where each platform has a distinct request completion time. I've derived these times from the average of potential failed counts within a minute, aiding in minimizing false positive events and reducing noise.

petebryan commented on Detections/MultipleDataSources/ExtensiveAWSComputeResurseDeployements.yaml:
```diff
@@ -0,0 +1,54 @@
+id: 9e457dc4-81f0-4d25-bc37-a5fa4a17946a
+name: Extensive AWS Compute Resurse Deployements
```
Could you also do a cross-cloud version of this where we see suspicious compute deployment in AWS and Azure in same time frame by same actor?

4R9UN replied:
We will activate this through correlation rules, grouping multiple IOAIDs and incorporating time difference conditions. This approach will streamline the process and enhance efficiency. However, this isn't necessary at the moment.

petebryan commented on Detections/MultipleDataSources/ExtensiveAWSComputeResurseDeployements.yaml:
```diff
@@ -0,0 +1,54 @@
+id: 9e457dc4-81f0-4d25-bc37-a5fa4a17946a
+name: Extensive AWS Compute Resurse Deployements
```
Suggest renaming to "Suspicious AWS EC2 Compute Resource Deployments"

4R9UN replied:
done

petebryan commented on Detections/MultipleDataSources/ExtensiveAWSComputeResurseDeployements.yaml:
```diff
@@ -0,0 +1,54 @@
+query: |
+  // Retrieve AWS CloudTrail events generated within the last day
+  AWSCloudTrail
+  | where TimeGenerated >= ago(1d)
```
Again not needed as it matched the query period value

4R9UN replied:
removed

petebryan commented on Detections/MultipleDataSources/ExtensiveAWSComputeResurseDeployements.yaml:
```diff
@@ -0,0 +1,54 @@
+  // Filter results based on total GPU compute and time duration
+  | where totalgpu > 800
+  | where datetime_diff('hour', end, Start) < 8
+entityMappings:
```
I would consider using alertDetailsOverride and customDetails here to further customize this alert.

Also is it possible to identity and project the account entities that deployed these resources?

4R9UN replied:
customDetails added, UserIdentityArn can be used identify the account user of the deployments.

petebryan commented on Detections/MultipleDataSources/ExtensiveAWSComputeResurseDeployements.yaml:
```diff
@@ -0,0 +1,54 @@
+  | extend OSplatform = iff(isempty(platform), tostring("Linux"), platform),CPU = tostring(parse_json(ResponseElements).instancesSet.items[0].cpuOptions),core = toint(parse_json(ResponseElements).instancesSet.items[0].cpuOptions.coreCount),corThread = toint(parse_json(ResponseElements).instancesSet.items[0].cpuOptions.threadsPerCore),InstanceId = tostring(parse_json(ResponseElements).instancesSet ...
+  // Calculate the total compute based on core and thread counts
+  | extend totalCorecompute = core * corThread
+  // Filter out instances with empty core values
```
How do you have instances without CPU cores?

4R9UN replied:
there are some instance where we are not able to find the core numbers, this could be telemetry gap, to reduce this kind of noise I added this conditions. I just edit it one line above in query.

petebryan commented on Detections/MultipleDataSources/SucessfullPosswordSprayAttampOnAWSConsolelogin.yaml:
```diff
@@ -0,0 +1,121 @@
+id: 188db479-d50a-4a9c-a041-644bae347d1f
+name: Successful Password Spray Attempt on AWS Console login
```
I would rename this to better reflect the activity - maybe something like "Successful AWS Console Login from IP Address Observed Conducting Password Spray"

4R9UN replied:
changed

petebryan commented on Detections/MultipleDataSources/SucessfullPosswordSprayAttampOnAWSConsolelogin.yaml:
```diff
@@ -0,0 +1,121 @@
+  - T1078
+query: |
+  SecurityAlert 
+    | where TimeGenerated >= ago(1d)
```
Matches queryPeriod

petebryan commented on Detections/MultipleDataSources/SucessfullPosswordSprayAttampOnAWSConsolelogin.yaml:
```diff
@@ -0,0 +1,121 @@
+        )
+        on $left.EntityIp == $right.SourceIpAddress 
+  // Filtering login event after the Alert generation time
+    | where AlertTimeGenerated >= AWSTime
```
Will this always happen? Sometimes products can take some time to generate alerts. You might be better taking the Alerts first activity time rather than the time the alert itself was generated.

petebryan commented on Detections/MultipleDataSources/SuspiciousAWSCLICommandExecution.yaml:
```diff
@@ -0,0 +1,47 @@
+query: |
+  // Retrieve AWS CloudTrail events
+  AWSCloudTrail 
+  | where TimeGenerated >= ago(1d)
```
Matches queryPeriod so not needed

petebryan commented on Detections/MultipleDataSources/SuspiciousAWSCLICommandExecution.yaml:
```diff
@@ -0,0 +1,47 @@
+  // Extract the command from the UserAgent using string splitting
+  | extend command = tostring(split(UserAgent, "off command/", 1)[0])  
+  // Filter events based on predefined suspicious command list
+  | where command in~ ('iam.list-users', 'iam.list-groups', 'ec2.describe-vpcs', 'ec2.describe-subnets', 'route53.list-hosted-zones', 'kms.list-keys', 'kms.list-aliases', 'ecs.list-clusters', 'ecs.list-services', 'iam.list-roles', 'iam.get-user', 'iam.list-access-keys', 'ec2.describe-security-groups', 'ec2.describe-network-acls', 'ec2.describe-network-interfaces', 'ec2.describe-route-tables', 'ec ...
```
Maybe defined these in a let statement at the top for easier reading.

petebryan commented on Detections/MultipleDataSources/ExtensiveAWSComputeResurseDeployements.yaml:
```diff
@@ -0,0 +1,54 @@
+id: 9e457dc4-81f0-4d25-bc37-a5fa4a17946a
```
This is in the MultipleDataSources folder but I only see it using CloudTrail, wouldn't it be better in the AWS solution?

petebryan commented on Detections/MultipleDataSources/SucessfullPosswordSprayAttampOnAWSConsolelogin.yaml:
```diff
@@ -0,0 +1,121 @@
+id: 188db479-d50a-4a9c-a041-644bae347d1f
```
Could we also do the reverse here? A GuardDuty alert for password spray identifying an IP Address logging into Azure?

4R9UN replied:
we do not have Guard Duty alerts in RDA, without RDA testing its risky to make detection

petebryan commented on Detections/MultipleDataSources/SuspiciousAWSConsolLoginByCredentialAceessAlerts.yaml:
```diff
@@ -0,0 +1,125 @@
+  - T1078
+query: |
+  SecurityAlert 
+    | where TimeGenerated >= ago(1d)
```
covered by queryPeriod

petebryan commented on Detections/MultipleDataSources/SuspiciousAWSConsolLoginByCredentialAceessAlerts.yaml:
```diff
@@ -0,0 +1,125 @@
+            AWSTime,
+            UserIdentityUserName
+        )
+        on $left.EntityIp == $right.SourceIpAddress 
```
Do we want to join on account name as well, that would seem like the stronger entity type for Credential Access alerts.

4R9UN replied:
Account name entity is different in both table unable to match. required OneIdentity.

petebryan commented on Detections/MultipleDataSources/UserImpersonateByAAID.yaml:
```diff
@@ -0,0 +1,67 @@
+query: |
+ // Retrieve SecurityAlerts generated within the last day
+  SecurityAlert 
+  | where TimeGenerated >= ago(1d)
```
Covered by queryPeriod

petebryan commented on Detections/MultipleDataSources/UserImpersonateByAAID.yaml:
```diff
@@ -0,0 +1,67 @@
+  | where ProductName has "Azure Active Directory Identity Protection"
+  | where AlertSeverity == "High"
+  // Extract IP address entities from the 'Entities' field
+  | extend ipAddress = extract(@'\b(?:\d{1,3}\.){3}\d{1,3}\b', 0, Entities)
```
Is this the most efficent method?

petebryan commented on Detections/MultipleDataSources/UserImpersonateByAAID.yaml:
```diff
@@ -0,0 +1,67 @@
+      AWSCloudTrail
+      | where TimeGenerated >= ago(1d)
+      | where isempty(ErrorMessage)
+      //|where EventSource == "iam.amazonaws.com"
```
Why is this commented out?

4R9UN replied:
we are expanding the coverage source, not limiting to IAM but it will also cover webmail and other Identity solutions

4R9UN replied:
if its generate any FP then we will narrow down the scope

petebryan replied:
In that case we should either remove completely or add an extra comment to explain this to people using this template. Currently it's a bit confusing.

4R9UN replied:
I have removed

petebryan commented on Detections/MultipleDataSources/UserImpersonateByAAID.yaml:
```diff
@@ -0,0 +1,67 @@
+  SecurityAlert 
+  | where TimeGenerated >= ago(1d)
+  // Filter alerts for Azure Active Directory Identity Protection and High severity
+  | where ProductName has "Azure Active Directory Identity Protection"
```
Why are we just limiting to IDP here and not other products as well?

4R9UN replied:
Identity related Alert are very accurate in IDP, and in the M365D has created the same detection for Azure with that reference and future correlation objectives I have created this detection only for IDP

petebryan commented on Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml:
```diff
@@ -0,0 +1,69 @@
+id: f4a28082-2808-4783-9736-33c1ae117475
+name: High-Risk Cross-Cloud User impersonation 
```
Capitalize the I in impersonation

petebryan replied:
Also not clear how this title reflects what the query is looking for

petebryan commented on Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml:
```diff
@@ -0,0 +1,69 @@
+query: |
+  // Retrieve Azure AD SigninLogs within the last day
+  SigninLogs 
+  | where TimeGenerated >= ago(1d)
```
Covered by queryPeriod

4R9UN replied:
removed

petebryan commented on Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml:
```diff
@@ -0,0 +1,69 @@
+id: f4a28082-2808-4783-9736-33c1ae117475
+name: High-Risk Cross-Cloud User impersonation 
+description: |
+  'This detection focuses on dentifying high-risk cross-cloud activities and sign-in anomalies that may indicate potential security threats. The query starts by analyzing Azure AD SigninLogs to pinpoint instances where specific applications, risk levels, and result types align. It then correlates this information with relevant AWS CloudTrail events to identify activities across Azure and AWS envi ...
```
Some typos in here

4R9UN replied:
fixed

petebryan commented on Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml:
```diff
@@ -0,0 +1,69 @@
+  | where TimeGenerated >= ago(1d)
+  // Filter for specific AppDisplayNames, ResultType, and Risk Levels
+  | where AppDisplayName in ("Azure Portal", "ADFS Trust", "Microsoft Azure PowerShell")
+      and ResultType == 0
```
Given we are joining on just IP why wouldn't we also want to capture risky failed logons?

4R9UN replied:
objective of this query is to detect the successful High risk SignIn and the find suspicious activity in AWS, for fil logon we can create another detection

petebryan replied:
I get that but wouldn't a failed high risk login attempt performing some action in AWS also be worth detecting? If I am seeing Azure blocking and IP due to some risk factor I would definitely want to know if that IP then starts doing stuff in AWS. I would even argue that this is more valuable that a high risk successful sign in from Azure as the success of that sign in suggests that the high risk tag may have been mitigated by the user.

4R9UN replied:
Got your point, I have removed the ResultType== 0 condition and added description of event as below.
| extend Result = iff(ResultType == 0, "Successful Signin", "Failed Signin")

petebryan commented on Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml:
```diff
@@ -0,0 +1,69 @@
+  | extend timedef = datetime_diff("hour", EventTime, signInTime)
+  // Filter for time differences within a certain range
+  | where timedef between (0 .. 8)
+entityMappings:
```
What about the User entity here?

4R9UN replied:
added in customDetails

petebryan replied:
I see that but why is it not also an Entity here? We use Entities for correlations and the custom details to add context. If we don't add user to entityMappings we can't use it for correlations.

4R9UN replied:
Actually, we are not able to add a user entity here as we do not have a use name separately in AWS, That is an issue we have raised that could be overcome by One Identity solution

petebryan commented on Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml:
```diff
@@ -0,0 +1,69 @@
+id: f4a28082-2808-4783-9736-33c1ae117475
```
Could we do another query with the reverse here as well, risky AWS signin seen in Azure?

4R9UN replied:
AWS doesn't have any Risky Signin signals

petebryan commented on Detections/MultipleDataSources/SuspiciousAWSCLICommandExecution.yaml:
```diff
@@ -0,0 +1,47 @@
+tactics:
+  - Reconnaissance
+relevantTechniques:
+  - T1596
```
Not sure I agree with this technique mapping

4R9UN replied:
Update : 
relevantTechniques:
  - T1595
  - T1592.004
  - T1589.002
  - T1589.003
  - T1590
  - T1591
  - T1596

petebryan commented on Detections/MultipleDataSources/UserImpersonateByAAID.yaml:
```diff
@@ -0,0 +1,67 @@
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1134
```
I don't see where Token Manipulation is being looked for here

4R9UN replied:
"CreateAccessKey", "DeleteAccessKey"can be used for token or key manipulation

petebryan commented on Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml:
```diff
@@ -0,0 +1,69 @@
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1134
```
Not seeing token manipulation

4R9UN committed:
Update BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml

Title Changed

4R9UN committed:
Update BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml

Update : 
-customDetails
- enabled Logic for both platform (AWS or Azure)

4R9UN committed:
Update and rename ExtensiveAWSComputeResurseDeployements.yaml to SuspiciousAWSEC2ComputeResourceDeployments

Update- 
-Changed Title
-added customDetails
-slight change in detection logic

4R9UN committed:
Update and rename SucessfullPosswordSprayAttampOnAWSConsolelogin.yaml to SuccessfulAWSConsoleLoginfromIPAddressObservedConductingPasswordSpray.yaml

-Name and Title Changed
-Timing logic changed

4R9UN committed:
Update SuspiciousAWSCLICommandExecution.yaml

Slight update in Query logic and added some relevantTechniques:

4R9UN committed:
Update SuspiciousAWSConsolLoginByCredentialAceessAlerts.yaml

removed the time period

4R9UN committed:
Update UserImpersonateByAAID.yaml

removed the time condition

4R9UN committed:
Update UserImpersonateByRiskyUser.yaml

Update with Custom details and removed the time filter

4R9UN reviewed (commented):
Addressed All the comments.

4R9UN committed:
Update BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml

4R9UN committed:
Update UserImpersonateByRiskyUser.yaml

4R9UN committed:
Update BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml

4R9UN committed:
Update SuccessfulAWSConsoleLoginfromIPAddressObservedConductingPasswordSpray.yaml

4R9UN committed:
Update SuspiciousAWSCLICommandExecution.yaml

4R9UN committed:
Update SuspiciousAWSConsolLoginByCredentialAceessAlerts.yaml

4R9UN committed:
Update SuspiciousAWSEC2ComputeResourceDeployments

4R9UN committed:
Update UserImpersonateByAAID.yaml

4R9UN committed:
Update UserImpersonateByRiskyUser.yaml

4R9UN reviewed (commented):
Fix Git error

4R9UN committed:
Update and rename SuspiciousAWSEC2ComputeResourceDeployments to SuspiciousAWSEC2ComputeResourceDeployments.yaml

4R9UN reviewed (commented):
file name changed

4R9UN committed:
Update SuspiciousAWSEC2ComputeResourceDeployments.yaml

4R9UN reviewed (commented):
Git Error Fixing

4R9UN committed:
2 Detection AWSGuardDuty to Azure Login 

Created 2 more detections for AWSGuardDuty to Azure portal Login Activity

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

PR Error Fix

4R9UN committed:
Update CrossCloudUnauthorizedCredentialsAccessDetection.yaml

PR error Fix

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

4R9UN committed:
Update CrossCloudUnauthorizedCredentialsAccessDetection.yaml

4R9UN reviewed (commented):
PR Error Fix

4R9UN committed:
Update CrossCloudUnauthorizedCredentialsAccessDetection.yaml

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

4R9UN reviewed (commented):
FIX PR Error

4R9UN committed:
Update CrossCloudUnauthorizedCredentialsAccessDetection.yaml

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

4R9UN reviewed (commented):
Fix PR error

4R9UN committed:
Update CrossCloudUnauthorizedCredentialsAccessDetection.yaml

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

4R9UN reviewed (commented):
Fix PR error

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

4R9UN reviewed (commented):
PR Error Fix

4R9UN commented:
@petebryan, I have updated 2 more detections for AWSGurdduty to the Azure portal login, kindly review.

petebryan commented on Detections/MultipleDataSources/BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml:
```diff
@@ -0,0 +1,102 @@
+  Azure_Bruforce
+  | join kind=inner AWS_Bruforce on $left.IPAddress == $right.SourceIpAddress and $left.UserAgent == $right.UserAgent
+  // Filtering based on conditions for failed and successful events
+  | where (AWSFailedEventsCount >= 4 and AWSSuccessfuleventsCount >= 1 and AWSFailedEventsCount > AWSSuccessfuleventsCount) or (AzureFailedEventsCount >= 5 and AzureSuccessfuleventsCount >= 1 and AzureFailedEvent > AzureSuccessfulEvent)
```
If we wanted to look for distributed attempts that targeted both clouds but were only successful in one cloud wouldn't want the logic to be:
| where (AWSFailedEventsCount >= 4 and AzureFailedEventsCount >= 5) and ((AzureSuccessfuleventsCount >= 1 and 
AzureFailedEvent > AzureSuccessfulEvent) or (AWSSuccessfuleventsCount >= 1 andAWSFailedEventsCount > AWSSuccessfuleventsCount))

4R9UN replied:
Updated As per your suggestion

petebryan commented on Detections/MultipleDataSources/CrossCloudUnauthorizedCredentialsAccessDetection.yaml:
```diff
@@ -0,0 +1,149 @@
+id: 122fbc6a-57ab-4aa7-b9a9-51ac4970cac1
+name: Cross-Cloud Unauthorized Credential Access Detection
```
It would be good to mention the RDS scope in the name as well - so when triaging analysts understand the RDS element

4R9UN replied:
Added

4R9UN replied:
Updated As per your suggestion

petebryan commented on Detections/MultipleDataSources/CrossCloudUnauthorizedCredentialsAccessDetection.yaml:
```diff
@@ -0,0 +1,149 @@
+  AzAuthRequirement: AuthenticationRequirement
+  alertSeverity: Severity
+alertDetailsOverride:
+  alertDisplayNameFormat: "{{AWSAlertTitle}}"
```
Do we completely want to replace the alert name with that? I would think it might be better to be:

"IP address {{IPAddress}} in {{AWSAlertTitle}} seen in Azure Signin Logs with {{UserPrincipalName}}"

4R9UN replied:
Updated As per your suggestion

petebryan commented on Detections/MultipleDataSources/SuspiciousAWSCLICommandExecution.yaml:
```diff
@@ -0,0 +1,57 @@
+id: 8c2dc344-9352-4ca1-8863-b1b7a5e09e59
```
Given this just uses AWS data it should go in the AWS solution folder rather than in the multiple data sources detection one

petebryan commented on Detections/MultipleDataSources/SuspiciousAWSEC2ComputeResourceDeployments.yaml:
```diff
@@ -0,0 +1,54 @@
+id: 9e457dc4-81f0-4d25-bc37-a5fa4a17946a
```
Given this just uses AWS data it should go in the AWS solution folder rather than in the multiple data sources detection one

4R9UN replied:
The file was moved to the AWS solution folder.

petebryan commented on Detections/MultipleDataSources/Unauthorized_user_access_across_AWS_and_Azure.yaml:
```diff
@@ -0,0 +1,147 @@
+  AzAuthRequirement: AuthenticationRequirement
+  alertSeverity: Severity
+alertDetailsOverride:
+  alertDisplayNameFormat: "{{AWSAlertTitle}}"
```
Suggest you update to something like:

"{{AWSNetworkEntity}} from {{AWSAlertTitle}} observed in Azure Singins with {{UserPrincipalName}}"

4R9UN replied:
Updated

petebryan commented on Detections/MultipleDataSources/Unauthorized_user_access_across_AWS_and_Azure.yaml:
```diff
@@ -0,0 +1,147 @@
+  alertSeverity: Severity
+alertDetailsOverride:
+  alertDisplayNameFormat: "{{AWSAlertTitle}}"
+  alertDescriptionFormat: "AWS Alert Description: '{{AWSAlertDescription}}' \n\n AWS ALert Link : '{{AWSAlertLink}}' \n\n Find More Details :https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_finding-types-active.html"
```
You want to combine this with details from the description field in this template as well as other fields in the data (such as key entities). The aim of this description field is to given the triaging analyst as much information as possible to understand what they are looking at.

4R9UN replied:
We have a limit of only 2 or 3 variables we can call here so we can not add another entity variable here, 2 are already called

petebryan replied:
It's not about the number of variables it's about the value this description brings to a user. If you saw this description as it is in an alert, would it provide all the details about to understand what the detection is looking for and what it has found? In its current state it doesn't do that.

4R9UN replied:
Ahh.. got your point, I just updated the alertDescriptionFormat details. kindly check

petebryan commented on Detections/MultipleDataSources/CrossCloudUnauthorizedCredentialsAccessDetection.yaml:
```diff
@@ -0,0 +1,149 @@
+  alertSeverity: Severity
+alertDetailsOverride:
+  alertDisplayNameFormat: "{{AWSAlertTitle}}"
+  alertDescriptionFormat: "AWS Alert Description: '{{AWSAlertDescription}}' \n\n AWS ALert Link : '{{AWSAlertLink}}' \n\n Find More Details :https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_finding-types-active.html"
```
You want to combine this with details from the description field in this template as well as other fields in the data (such as key entities). The aim of this description field is to given the triaging analyst as much information as possible to understand what they are looking at.

4R9UN replied:
We have a limit of only 2 or 3 variables we can call here so we can not add another entity variable here, 2 already called

petebryan replied:
It's not about the number of variables it's about the value this description brings to a user. If you saw this description as it is in an alert, would it provide all the details about to understand what the detection is looking for and what it has found? In its current state it doesn't do that.

4R9UN replied:
Ahh.. got your point, I just updated the alertDescriptionFormat details. kindly check

4R9UN committed:
Update BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml

Accepted suggested logic 

| where (AWSFailedEventsCount >= 4 and AzureFailedEventsCount >= 5) and ((AzureSuccessfuleventsCount >= 1 and AzureFailedEvent > AzureSuccessfulEvent) or (AWSSuccessfuleventsCount >= 1 and AWSFailedEventsCount > AWSSuccessfuleventsCount))

4R9UN committed:
Update CrossCloudUnauthorizedCredentialsAccessDetection.yaml

Updated AWS Title

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

4R9UN committed:
Added New Detection for AWS cloud Shell execution 

Added New Detection for AWS cloud Shell execution

4R9UN reviewed (commented):
Updated all the changes

4R9UN committed:
Delete Detections/MultipleDataSources/SuspiciousAWSEC2ComputeResourceDeployments.yaml

File moved

4R9UN committed:
Delete Detections/MultipleDataSources/SuspiciousAWSCLICommandExecution.yaml

file moved

4R9UN committed:
Update UserImpersonateByAAID.yaml

Line removed

4R9UN committed:
Update UserImpersonateByRiskyUser.yaml

4R9UN committed:
Update CrossCloudUnauthorizedCredentialsAccessDetection.yaml

up[dated alertDescriptionFormat

4R9UN committed:
Update Unauthorized_user_access_across_AWS_and_Azure.yaml

4R9UN reviewed (commented):
Update files based on suggestions.

stesell commented on Detections/MultipleDataSources/BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml:
Why somebody would Bruteforce 
      | where AppDisplayName == "Azure Portal"
only ?

4R9UN replied:
possible trying to targeting all cloud portals to check the authentications
